File: .agents/skills/create-contract-python/evals/good/contract_file_system_port.py
from abc import ABC, abstractmethod
from shared.file_system.taxonomy_file_content_vo import FileContent
from shared.file_system.taxonomy_file_path_vo import FilePath
from shared.file_system.taxonomy_file_read_error import FileReadError


class IFileSystemPort(ABC):
    @abstractmethod
    def read_file(self, path: FilePath) -> Result[FileContent, FileReadError]: ...

File: .agents/skills/create-contract-typescript/evals/good/contract_file_system_port.ts
import { FileContent } from '../file_system/taxonomy_file_content_vo';
import { FilePath } from '../file_system/taxonomy_file_path_vo';
import { FileReadError } from '../file_system/taxonomy_file_read_error';

export interface IFileSystemPort {
    readFile(path: FilePath): Promise<Result<FileContent, FileReadError>>;
}
